google-cloud-storage/lib/google/cloud/storage/project.rb

```diff
@@ -94,6 +94,39 @@ def service_account_email
             service.project_service_account.email_address
         end
 
+        ##
+        # Add custom Google extension headers to the requests that use the signed URLs.
+        #
+        # @param [Hash] headers Google extension headers (custom HTTP headers that
+        #   begin with `x-goog-`) to be included in requests that use the signed URLs.
+        #   Provide headers as a key/value array, where the key is
+        #   the header name, and the value is an array of header values.
+        #   For headers with multiple values, provide values as a simple
+        #   array, or a comma-separated string. For a reference of allowed
+        #   headers, see [Reference Headers](https://cloud.google.com/storage/docs/xml-api/reference-headers).
+        #
+        # @return [Hash] Hash of Google extension headers included in requests that use the signed URLs.
```

Comment:
I think it might be better not to return this hash, as it's an internal hash (i.e. it's the current request options headers), and it would be better not to expose it in the interface or encourage callers to mess with it. Instead, for "modify options" methods like this, a good practice is to return `self` so that callers can "chain" method calls together.

Same for the `add_custom_header` call below.

Reply:
Thanks Daniel, will remember the method chaining concept for future as well